components/messaging/ConversationCardModal.tsx

```diff
@@ -0,0 +1,81 @@
+'use client';
+import React from 'react';
+import { useState } from 'react';
+import '../../app/styles/messaging-styles.css';
+import deleteConversation from '@/utils/messaging/deleteConversation';
+
+//Components
+import CardMenu from './CardMenu';
+import ButtonRounded from '../buttons/ButtonRounded';
+
+interface ModalProps {
+  conversationId?: number;
+  message: string;
+}
+
+const ConversationCardModal = ({ conversationId, message }: ModalProps) => {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [error, setError] = useState('');
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  const toggleModalClickHandler = (
+    event: React.MouseEvent<HTMLButtonElement>
+  ) => {
+    event.stopPropagation();
+    setModalOpen((prevState: boolean) => !prevState);
+  };
+
+  const deleteConversationClickHandler = async (
```

Comment:
I'm not as set on this one, but should we include the event on our Handler names or just the action performed?

`deleteConversationHandler` instead of `deleteConversationClickHandler` ?

The handlers could theoretically be re-used in other events and it makes the names a bit shorter and simpler without resorting to abbreviations.